src/applications/burials/BurialsApp.jsx

```diff
@@ -3,41 +3,65 @@ import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
 import RoutedSavableApp from 'platform/forms/save-in-progress/RoutedSavableApp';
-import { useFeatureToggle } from 'platform/utilities/feature-toggles';
 import { toggleValues } from 'platform/site-wide/feature-toggles/selectors';
+import { VA_FORM_IDS } from 'platform/forms/constants';
 import formConfig from './config/form';
 import { NoFormPage } from './components/NoFormPage';
 
-function BurialsEntry({ location, children, isLoadingFeatures }) {
-  const { useToggleValue, TOGGLE_NAMES } = useFeatureToggle();
-  const burialFormEnabled = useToggleValue(TOGGLE_NAMES.burialFormEnabled);
-  const redirectToHowToPage =
-    burialFormEnabled === false && location.pathname !== '/introduction';
-  if (redirectToHowToPage === true) {
-    window.location.href = '/burials-memorials/veterans-burial-allowance/';
-  }
-
-  if (isLoadingFeatures !== false || redirectToHowToPage) {
+function BurialsEntry({ location, children, toggleFeatures, profile }) {
+  const {
+    loading: isLoadingFeatures,
+    burialFormEnabled,
+    burialFormV2,
+  } = toggleFeatures;
+  if (isLoadingFeatures) {
     return <va-loading-indicator message="Loading application..." />;
   }
 
   if (!burialFormEnabled) {
+    if (location.pathname !== '/introduction') {
+      window.location.href = '/burials-memorials/veterans-burial-allowance/';
+    }
     return <NoFormPage />;
   }
+
+  const hasV1Form = profile.savedForms.some(form => {
+    // Two is the correct metadata version.
+    return (
+      form.form === VA_FORM_IDS.FORM_21P_530 && form.metadata.version === 2
```

Comment:
if we extract `2` and `3` in to 
`const 2017MetadataVersion = 2`
`const 2022MetadataVersion = 3`

it will prevent confusion here.